tests/test_datacite.py

```diff
@@ -14,106 +14,110 @@ def test_jpal_fixtures(datacite_jpal_records):
+                affiliation=["Abdul Latif Jameel Poverty Action Lab"],
+                identifier=None,
+                kind="Creator",
+                mit_affiliated=None,
+            ),
+            Contributor(
+                value="Banerji, Rukmini",
+                affiliation=["Pratham and ASER Centre"],
+                identifier=["https://orcid.org/0000-0000-0000-0000"],
+                kind="ContactPerson",
+                mit_affiliated=None,
+            ),
+        ],
+        dates=[Date(kind="Publication date", note=None, range=None, value="2017")],
+        identifiers=[Identifier(value="10.7910/DVN/19PPE7", kind="DOI")],
+        notes=[Note(value=["Survey Data"], kind="Datacite resource type")],
+        publication_information=["Harvard Dataverse"],
+    )
 
 
-def test_datacite_minimal_record(datacite_record, datacite_jpal_record_minimal):
-    datacite_record.input_records = datacite_jpal_record_minimal
-    output_records = datacite_record
-    for output_record in output_records:
-        assert output_record == TimdexRecord(
-            source="A Cool Repository",
-            source_link="https://example.com/item123doi:10.7910/DVN/19PPE7",
-            timdex_record_id="cool:doi:10.7910-DVN-19PPE7",
-            title="The Impact of Maternal Literacy and Participation Programs",
-            content_type=None,
-            contributors=None,
-            dates=None,
-            identifiers=[Identifier(value="10.7910/DVN/19PPE7", kind="DOI")],
-            notes=None,
-            publication_information=None,
-        )
+def test_datacite_required_fields_record(
+    datacite_record_partial, datacite_jpal_record_required_fields
+):
+    output_records = datacite_record_partial(
+        input_records=datacite_jpal_record_required_fields
+    )
+    assert next(output_records) == TimdexRecord(
+        source="A Cool Repository",
+        source_link="https://example.com/item123doi:10.7910/DVN/19PPE7",
+        timdex_record_id="cool-repo:doi:10.7910-DVN-19PPE7",
+        title="The Impact of Maternal Literacy and Participation Programs",
+        content_type=None,
+        contributors=None,
+        dates=None,
+        identifiers=[Identifier(value="10.7910/DVN/19PPE7", kind="DOI")],
+        notes=None,
+        publication_information=None,
+    )
 
 
 def test_datacite_missing_required_field(
-    datacite_record, datacite_jpal_record_missing_required_field
+    datacite_record_partial, datacite_jpal_record_missing_required_field
 ):
     with raises(ValueError):
-        datacite_record.input_records = datacite_jpal_record_missing_required_field
-        output_records = datacite_record
-        for output_record in output_records:
-            assert output_record
+        output_records = datacite_record_partial(
+            input_records=datacite_jpal_record_missing_required_field
+        )
+        assert next(output_records)
 
 
 def test_datacite_multiple_titles(
```

Comment:
These tests should also be named to indicate what they are testing (raises error).